flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/RecordWriter.java

```diff
@@ -276,6 +281,21 @@ protected void checkErroneous() throws IOException {
 		}
 	}
 
+	protected void addBufferConsumer(BufferConsumer consumer, int targetChannel) throws IOException {
+		targetPartition.addBufferConsumer(consumer, targetChannel);
+	}
+
+	protected BufferBuilder getBufferBuilder() throws IOException, InterruptedException {
+		BufferBuilder builder;
+		if (!targetPartition.isApproximatelyAvailable()) {
+			long start = System.currentTimeMillis();
+			builder = targetPartition.getBufferBuilder();
+			idleTimeMsPerSecond.markEvent(System.currentTimeMillis() - start);
+		} else {
+			builder = targetPartition.getBufferBuilder();
+		}
+		return builder;
+	}
```

Comment:
`NetworkBufferPool#redistributeBuffers` reduce the buffer available only when new task deployed right? I think it can be ignored, because we use meter to measure average idle time which use a time window by default 60s, so the side effect can be ignored.

Reply:
The thing is that the code might change/evolve over time making this check even more wrong. Yes now this bug is pretty harmless, but if we can make it easily completely correct, why not doing it? Proper fix is just ~5 lines of code in `LocalBufferPool` to provide `LocalBufferPool#requestBufferBuilder`, maybe 5 more lines `TestPooledBufferProvider`, add the method to the `BufferPool` interface and proxy/expose it in the `ResultPartition`.

Reply:
Thanks for the update!